Copilot/find instances of cite (#294)
symbiotic-engineering/MDOcean: pubs/renewable-energy-mdo/sections/optimization.tex

\ifdefined\DISSERTATION
    First, design space exploration is performed using a one-at-a-time (OAT) sweep, in which each design variable is individually varied while the others remain at their nominal values.
    This provides an initial understanding of the problem behavior, can reveal objective multimodality, and shows whether the objective and constraints are locally monotonic in each variable.

>     OAT does not capture the effect of coupling between design variables, which methods such as orthogonal arrays would expose \citet{box_statistics_1978}, but the cheapness of the simulation incentivizes moving directly to optimization rather than pursuing more thorough exploration.

Author: ```suggestion
    OAT does not capture the effect of coupling between design variables, which methods such as orthogonal arrays would expose \citep{box_statistics_1978}, but the cheapness of the simulation incentivizes moving directly to optimization rather than pursuing more thorough exploration.
```

\else
    First, design space exploration uses a one-at-a-time (OAT) sweep, varying each design variable individually while holding the others at nominal values to reveal local monotonicity and potential multimodality.

>     OAT does not capture inter-variable coupling \citet{box_statistics_1978}, but the cheap simulation incentivizes proceeding directly to optimization.

Author: ```suggestion
    OAT does not capture inter-variable coupling \citep{box_statistics_1978}, but the cheap simulation incentivizes proceeding directly to optimization.
```

As a gradient-based optimizer, SQP requires derivatives of the objective and constraints with respect to the design variables.
\ifdefined\DISSERTATION
    Obtaining derivatives for WEC simulations is a difficult problem that few have approached.

>     WecOptTool uses the Python package JAX \citet{bradbury_jax_2018} to automatically differentiate the pseudo-spectral dynamics, but cannot differentiate the hydrodynamic coefficients.

Author: ```suggestion
    WecOptTool uses the Python package JAX \citep{bradbury_jax_2018} to automatically differentiate the pseudo-spectral dynamics, but cannot differentiate the hydrodynamic coefficients.
```

>     Two recent papers introduce derivatives for BEM hydrodynamics packages \citep{rohrer_analytical_2024,khanal_fully_2025}, the former reviewing other work on hydrodynamic gradients.

Author: ```suggestion
    Recently, \citet{rohrer_analytical_2024,khanal_fully_2025} introduce derivatives for BEM hydrodynamics packages, and the former reviews other work on hydrodynamic gradients.
```

    Most similar to the present approach, \citet{gambarini_gradient_2024} uses semi-analytical hydrodynamics and analytically differentiates the linear system with respect to control and array-layout variables, but does not provide the body-dimension derivatives required here.
    For the MEEM hydrodynamics used in this study, symbolic differentiation of the linear system was attempted but exhausts memory at any reasonable matrix size; obtaining MEEM gradients via adjoints or algorithmic differentiation remains future work.
    This study therefore uses finite difference, with step size set automatically by the MATLAB Optimization Toolbox --- a viable choice given the simulation's low cost.

>     Initial concerns that the oscillatory hydrodynamics would degrade the gradients \citet{mccabe_open-source_2024} proved unfounded, likely because the overall hydrodynamic coefficients are far less oscillatory than their individual subfunctions, keeping gradients accurate even at larger step sizes.

Author: ```suggestion
    Initial concerns that the oscillatory hydrodynamics would degrade the gradients \citep{mccabe_open-source_2024} proved unfounded, likely because the overall hydrodynamic coefficients are far less oscillatory than their individual subfunctions, keeping gradients accurate even at larger step sizes.
```

\else

>     Obtaining gradients for WEC simulations is difficult: existing automatic and analytical differentiation methods cover pseudo-spectral dynamics \citet{bradbury_jax_2018} or BEM and semi-analytical hydrodynamics \citet{rohrer_analytical_2024,khanal_fully_2025,gambarini_gradient_2024} but none supply the body-dimension derivatives of the MEEM hydrodynamics used here, and symbolic differentiation exhausts memory.

Author: ```suggestion
    Obtaining gradients for WEC simulations is difficult: existing automatic and analytical differentiation methods cover pseudo-spectral dynamics \citep{bradbury_jax_2018} or BEM and semi-analytical hydrodynamics \citep{rohrer_analytical_2024,khanal_fully_2025,gambarini_gradient_2024} but none supply the body-dimension derivatives of the MEEM hydrodynamics used here, and symbolic differentiation exhausts memory.
```

>     This study therefore uses finite difference, viable given the cheap simulation; initial concerns about oscillatory hydrodynamics \citet{mccabe_open-source_2024} proved unfounded, as the overall coefficients are far less oscillatory than their subfunctions.

Author: ```suggestion
    This study therefore uses finite difference, viable given the cheap simulation; initial concerns about oscillatory hydrodynamics \citep{mccabe_open-source_2024} proved unfounded, as the overall coefficients are far less oscillatory than their subfunctions.
```

    These sensitivities estimate not only how the objective would change if a parameter changed at fixed design, but also how the optimal objective, optimal design, and constraint activity would change if the optimization were repeated with the modified parameter, without re-optimizing, and thus at very little cost.
    They serve as a screening step to identify high-sensitivity parameters for further study in the global sensitivity analysis.
\else

>     Local sensitivities exploit the SQP solver's Lagrange multiplier vector $\vec{\lambda} = -dJ/d\vec{g}$ \citet{martins_engineering_2022}, which measures the objective's sensitivity to each constraint; nonzero multipliers mark active constraints, larger values more influential ones.

Author: ```suggestion
    Local sensitivities exploit the SQP solver's Lagrange multiplier vector $\vec{\lambda} = -dJ/d\vec{g}$ \citep{martins_engineering_2022}, which measures the objective's sensitivity to each constraint; nonzero multipliers mark active constraints, larger values more influential ones.
```

    Two more common approaches exist: reformulating the multi-objective problem as repeated single-objective optimization to permit a gradient-based algorithm, or using a multi-objective derivative-free or heuristic method that extends readily from its single-objective counterpart.
    This work pursues both: the epsilon-constraint method first creates an ensemble of single-objective problems solved by the familiar SQP algorithm, and these solutions then seed a derivative-free pattern search.
    In the epsilon-constraint method, one objective is repeatedly optimized while a constraint on the second objective is swept, tracing out the Pareto front.
    Here the single-objective optima of each objective set the constraint values: the nadir point $(J_1(J_2=J_2^*),\, J_2(J_1=J_1^*))$ becomes the center of a circle, and twenty constraint values are taken at equal angular spacing on that circle.

>     The MATLAB \texttt{paretosearch} function then executes the constrained multi-objective pattern search of \citep{custodio_direct_2011}, starting from the SQP-optimized epsilon-constraint points and maximizing Pareto hypervolume and spread to quickly fill gaps between them.

Author: ```suggestion
    The MATLAB \texttt{paretosearch} function then executes a constrained multi-objective pattern search \citep{custodio_direct_2011}, starting from the SQP-optimized epsilon-constraint points and maximizing Pareto hypervolume and spread to quickly fill gaps between them.
```

\else
    While derivative criteria for local Pareto optimality exist \citep{desideri_multiple-gradient_2012}, solvers using them are rare, since a Pareto front demands exploration rather than the pure exploitation that gradient methods excel at.

>     This work combines two approaches: the epsilon-constraint method recasts the problem as an ensemble of single-objective problems solved by SQP --- repeatedly optimizing one objective while sweeping a constraint on the other, with twenty constraint values placed at equal angular spacing on a circle centered at the nadir point $(J_1(J_2=J_2^*),\, J_2(J_1=J_1^*))$ --- and these solutions then seed the MATLAB \texttt{paretosearch} derivative-free pattern search \citet{custodio_direct_2011}, which maximizes Pareto hypervolume and spread to fill the gaps.

Author: ```suggestion
    This work combines two approaches: the epsilon-constraint method recasts the problem as an ensemble of single-objective problems solved by SQP --- repeatedly optimizing one objective while sweeping a constraint on the other, with twenty constraint values placed at equal angular spacing on a circle centered at the nadir point $(J_1(J_2=J_2^*),\, J_2(J_1=J_1^*))$ --- and these solutions then seed the MATLAB \texttt{paretosearch} derivative-free pattern search \citep{custodio_direct_2011}, which maximizes Pareto hypervolume and spread to fill the gaps.
```
